src/contracts/ODSaviour.sol

```diff
@@ -78,23 +78,22 @@ contract ODSaviour is Authorizable, Modifiable, ModifiablePerCollateral, IODSavi
     }
     if (!_enabledVaults[_vaultId]) revert VaultNotAllowed(_vaultId);
 
-    IOracleRelayer.OracleRelayerCollateralParams memory _oracleParams = oracleRelayer.cParams(_cType);
-    IDelayedOracle _oracle = _oracleParams.oracle;
-
     uint256 _reqCollateral;
     {
+      ISAFEEngine.SAFEEngineCollateralData memory _safeEngCData = safeEngine.cData(_cType);
+
       (uint256 _currCollateral, uint256 _currDebt) = getCurrentCollateralAndDebt(_cType, _safe);
-      uint256 _accumulatedRate = safeEngine.cData(_cType).accumulatedRate;
+      uint256 _accumulatedRate = _safeEngCData.accumulatedRate;
+      uint256 _liquidationPrice = _safeEngCData.liquidationPrice;
+      uint256 _safetyPrice = _safeEngCData.safetyPrice;
 
-      uint256 _currCRatio = ((_currCollateral.wmul(_oracle.read())).wdiv(_currDebt.wmul(_accumulatedRate)));
-      uint256 _safetyCRatio = _oracleParams.safetyCRatio / 1e18;
+      uint256 _collatXliqPrice = _currCollateral.wmul(_liquidationPrice);
+      uint256 _debtXaccumuRate = _currDebt.wmul(_accumulatedRate);
 
-      if (_safetyCRatio > _currCRatio) {
-        uint256 _diffCRatio = _safetyCRatio.wdiv(_currCRatio);
-        _reqCollateral = (_currCollateral.wmul(_diffCRatio)) - _currCollateral;
-      } else {
-        revert SafetyRatioMet();
-      }
+      _reqCollateral = (_debtXaccumuRate - _collatXliqPrice).wdiv(_safetyPrice);
+      uint256 _newCollatXliqPrice = (_reqCollateral + _currCollateral).wmul(_liquidationPrice);
+
+      if (_newCollatXliqPrice <= _debtXaccumuRate) revert SafetyRatioNotMet();
```

Comment:
rename 
- `_reqCollateral` to `_requiredCollateral` 
- `_collateralXliqPrice` to `_collateralXliquidationPrice`
- `_debtXaccumuRate` to `_debtXaccumulatedRate`
- `_newCollatXliqPrice` to `_newCollateralXliquidationPrice`

Reply:
Also change in the tests